### PR TITLE
Fix RecentDatabases

### PR DESCRIPTION
**Pull request by TheZ3ro**
## Description

Small fixes in the Welcome Screen.
When in Welcome screen and Recent Databases is cleared from the menu refresh the list.
Use the ItemActivated event to catch both double clicks and Enter keypress

## How has this been tested?

Manually

## Types of changes

- ✅ Bug fix (non-breaking change which fixes an issue)

## Checklist:

- ✅ I have read the **CONTRIBUTING** document. **[REQUIRED]**
- ✅ My code follows the code style of this project. **[REQUIRED]**
- ✅ All new and existing tests passed. **[REQUIRED]**
- ✅ I have compiled and verified my code with `-DWITH_ASAN=ON`. **[REQUIRED]**

### CODE REVIEW

File: src/gui/MainWindow.cpp [hunk 1]
```diff
@@ -378,6 +378,11 @@ void MainWindow::openRecentDatabase(QAction* action)
 void MainWindow::clearLastDatabases()
 {
     config()->set("LastDatabases", QVariant());
+    bool inWelcomeWidget = (m_ui->stackedWidget->currentIndex() == 2);
```
Comment (droidmonkey) [inline src/gui/MainWindow.cpp]: I've never been a fan of hardcoded index values like this. Much prefer defines to be more explicit. What do you think?

Comment (TheZ3ro) [inline src/gui/MainWindow.cpp]: Me too but I will have to change other places in the code where index are used

Comment (droidmonkey) [inline src/gui/MainWindow.cpp]: We can look to revamp this later

**Comment by louib** [comment]
@TheZ3ro needs a rebase!

**Comment by TheZ3ro** [comment]
@louib @droidmonkey rebased and ready to merge!
I will revamp the stackedWidget index soon

**Pull request merged**